pkg/discovery/dtofactory/general_builder.go

```diff
@@ -209,19 +216,29 @@ func (builder generalBuilder) metricValue(entityType metrics.DiscoveredEntityTyp
 	}
 
 	value := metric.GetValue()
-	switch value.(type) {
+	switch typedValue := value.(type) {
 	case []metrics.Point:
 		var sum float64
 		var peak float64
-		for _, point := range value.([]metrics.Point) {
+		for _, point := range typedValue {
 			sum += point.Value
 			peak = math.Max(peak, point.Value)
 		}
-		metricValue.Avg = sum / float64(len(value.([]metrics.Point)))
+		metricValue.Avg = sum / float64(len(typedValue))
 		metricValue.Peak = peak
+	case []metrics.ThrottlingCumulative:
+		numberOfSamples := len(typedValue)
+		if numberOfSamples <= 1 {
+			// We don't have enough samples to calculate this value.
+			// Throttling value would appear as zero on the entity.
+			break
+		}
+		metricValue.Avg = (typedValue[numberOfSamples-1].Throttled - typedValue[0].Throttled) /
+			(typedValue[numberOfSamples-1].Total - typedValue[0].Total)
+		metricValue.Peak = metricValue.Avg
```

Comment:
Probably can be an enhancement; if we have more than one sample, theoretically we could calculate the peak like in your original implementation.

Reply:
Calculating peak for containers and aggregating peaks on container specs.